File: 2271-RearrangeArrayElementsBySign/2271-RearrangeArrayElementsBySign.py
# Last updated: 6/11/2026, 11:23:39 AM
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        nums2 = [0 for i in range(len(nums))]
        k1= k2 = 0
        for i in range(0,len(nums)):
            if(nums[i]<0):
                nums2[2*k1+1] = nums[i]
                k1 +=1
            else:
                nums2[2*k2] = nums[i]
                k2 += 1
        return nums2

            

File: 2271-RearrangeArrayElementsBySign/2271-RearrangeArrayElementsBySign.cpp
// Last updated: 6/11/2026, 11:23:32 AM
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        queue<int> pos, neg;
        for(int e: nums){
            if(e>0){
                pos.push(e);
            }else{
                neg.push(e);
            }
        }

        vector<int> ans;
        for(int i=0; i<nums.size(); i++){
            if(i%2 == 0){
                int el = pos.front(); pos.pop();
                ans.push_back(el);
            }else{
                int el = neg.front(); neg.pop();
                ans.push_back(el);
            }
        }
        return ans;
    }
};